homeassistant/components/mqtt/discovery.py

```diff
@@ -48,7 +48,7 @@ def async_device_message_received(topic, payload, qos):
             return
 
         payload = dict(payload)
-        payload[CONF_PLATFORM] = 'mqtt'
+        payload[CONF_PLATFORM] = payload.get(CONF_PLATFORM, 'mqtt')
```

Comment:
Make sure we whitelist which platforms are allowed. We should not allow MQTT discovery to be used to launch any arbitrary platform.